scikeras/utils/__init__.py

```diff
@@ -0,0 +1,196 @@
+from inspect import isclass
+from typing import Callable, Union
+
+import numpy as np
+
+from sklearn.utils.multiclass import type_of_target as _type_of_target
+from tensorflow.keras.losses import Loss
+from tensorflow.keras.losses import get as get_loss
+from tensorflow.keras.losses import serialize as serialize_loss
+from tensorflow.keras.metrics import Metric
+from tensorflow.keras.metrics import get as get_metric
+from tensorflow.keras.metrics import serialize as serialize_metric
+
+
+def loss_name(loss: Union[str, Loss, Callable]) -> str:
```

Comment:
I don't think so. We don't use it recursively anywhere. I think if we did at some point, we could make a `loss_names` or something. I'm not a fan of implementing something until it's actually needed (or requested by users).

Reply:
I also think the error handling in `{loss,metric}_name` can be handled better. I might raise a `NotImplementedError` is lists/dicts are provided (instead of a ValueError?).

Reply:
I think if anything that would be a `TypeError`.  But do we really want to check the types here? I still feel that it's better to have a dead-simple API here where it either succeeds or raises a singular error that users can easily catch.

Reply:
If we were to do checks, I think it would be:
```python3
if not (isinstance(loss, (str, Loss)) or callable(loss)):
    raise TypeError(...)
```

Reply:
I think errors of the correct type should be raised (so I think a TypeError should be raised on the last line of https://github.com/adriangb/scikeras/pull/88#discussion_r502826143).

Why would `loss()` raise a TypeError? Why not remove the try-except block and let `loss()` raise it's own errors?

Reply:
> Why would `loss()` raise a TypeError?

Primarily it's called with missing arguments.

> Why not remove the try-except block and let `loss()` raise it's own errors?

Luckily I already pusshed https://github.com/adriangb/scikeras/pull/88/commits/2a37671fdf663d42b72e24e6df16a4b3a37f7cbe which does precisely that.

>  I think a TypeError should be raised on the last line of https://github.com/adriangb/scikeras/pull/88#discussion_r502826143

Well that won't be hard to implement, I'll add it as described in https://github.com/adriangb/scikeras/pull/88#discussion_r502826143

Reply:
https://github.com/adriangb/scikeras/pull/88#discussion_r502826143 is implemented. I'm going to resolve this thread since we have several others around the same topic.